src/Lecture3.hs

```diff
@@ -205,12 +203,12 @@ together only different elements.
 Product {getProduct = 6}
 
 -}
-appendDiff3 :: ( Eq a) =>[a] -> [a] -> [a] -> [a]
+appendDiff3 :: (Eq a, Monoid a) =>a -> a -> a -> a
 appendDiff3 a b c = a <> bunique <> cunique
    where bunique | a /= b = b
-                 | otherwise = mempty `asTypeOf` b
+                 | otherwise = mempty
          cunique | a /= c && b /= c  = c
-                 | otherwise = mempty `asTypeOf` c
+                 | otherwise = mempty
```

Comment:
That's a clever and concise implementation! 👏🏻 

However, the type signature is more strict than required. Your function works with `Monoid`. But not every type can have a `Monoid` instance. This means, that `appendDiff3` can't work with types like `List1`. But if you change the constraint from `Monoid` to `Semigroup`, this function will now work. Though, relaxing the constraint will require changing the implementation as well 🚧

Reply:
took sometime but couldn't figure it out

I will try after getting some sleep :sweat_smile: 

thanks again!